### PR TITLE
[Onprem] Add cluster config schema check

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Adds a scheme check for cluster config YAML for the Sky Onprem features

## Tests
- [x] Sky Onprem Smoke Tests

### CODE REVIEW

File: sky/utils/schemas.py [hunk 1]
```diff
@@ -155,3 +155,43 @@ def get_task_schema():
             },
         }
     }
+
+
+def get_cluster_schema():
+    return {
+        '$schema': 'https://json-schema.org/draft/2020-12/schema',
+        'type': 'object',
+        'required': [],
```
Comment (Michaelvll) [inline sky/utils/schemas.py:164]: Should we use `'required': ['cluster', 'auth']`, as they are required?

File: sky/utils/schemas.py [hunk 2]
```diff
@@ -155,3 +155,43 @@ def get_task_schema():
...
+        'additionalProperties': False,
+        'properties': {
+            'cluster': {
+                'type': 'object',
+                'required': [],
```
Comment (Michaelvll) [inline sky/utils/schemas.py:169]: Should we use `'required': ['ips', 'name']`?

File: sky/utils/schemas.py [hunk 3]
```diff
@@ -155,3 +155,43 @@ def get_task_schema():
...
+                        }
+                    },
+                    'name': {
+                        'type': 'string',
+                    },
+                }
+            },
+            'auth': {
+                'type': 'object',
+                'required': [],
```
Comment (Michaelvll) [inline sky/utils/schemas.py:185]: Should we use `'required': ['ssh_user', 'ssh_private_key']`?

**Review by Michaelvll — COMMENTED** [review]
Thank you for adding this @michaelzhiluo ! Left several comments for the schema. As all the fields are required for the on-prem cluster, should we put them in the `required`?

File: sky/utils/schemas.py (hunk 3 above)
Comment (michaelzhiluo) [reply sky/utils/schemas.py:185]: Done, thanks!

File: sky/utils/schemas.py (hunk 2 above)
Comment (michaelzhiluo) [reply sky/utils/schemas.py:169]: Done, thanks!

File: sky/utils/schemas.py (hunk 1 above)
Comment (michaelzhiluo) [reply sky/utils/schemas.py:164]: Done, thanks!

File: sky/backends/onprem_utils.py [hunk 4]
```diff
@@ -73,15 +75,20 @@ def check_and_get_local_clusters(suppress_error: bool = False) -> List[str]:
...
-                raise ValueError('Authentication into local cluster requires '
-                                 'specifying `ssh_user` and `ssh_private_key` '
-                                 'under the `auth` dictionary. Please fill '
-                                 'aforementioned fields in '
-                                 f'{sky_local_path.format(cluster_name)}.')
+
+        if not suppress_error:
+            backend_utils.validate_schema(yaml_config,
+                                          schemas.get_cluster_schema(),
+                                          'Invalid cluster YAML: ')
```
Comment (romilbhardwaj) [inline sky/backends/onprem_utils.py:82]: Should we do this validation _before_ L75 where we extract values from keys? Those lines may fail with KeyError if the yaml is malformed

File: sky/backends/onprem_utils.py [hunk 5]
```diff
@@ -73,15 +75,20 @@ def check_and_get_local_clusters(suppress_error: bool = False) -> List[str]:
...
-                not suppress_error):
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError('Authentication into local cluster requires '
-                                 'specifying `ssh_user` and `ssh_private_key` '
-                                 'under the `auth` dictionary. Please fill '
-                                 'aforementioned fields in '
-                                 f'{sky_local_path.format(cluster_name)}.')
+
+        if not suppress_error:
+            backend_utils.validate_schema(yaml_config,
```
Comment (romilbhardwaj) [inline sky/backends/onprem_utils.py:80]: Should we also do this validation when reading a YAML in `admin_deploy()` in cli.py?

**Review by romilbhardwaj — COMMENTED** [review]
Awesome! Great to have these validation checks!

File: sky/backends/onprem_utils.py [hunk 6]
```diff
@@ -65,6 +66,7 @@ def check_and_get_local_clusters(suppress_error: bool = False) -> List[str]:
 
     local_cluster_names = []
     name_to_path_dict = {}
+
     for path in local_cluster_paths:
         # TODO(mluo): Define a scheme for cluster config to check if YAML
         # schema is correct.
```
[the remark this answers is not in the document]
Comment (michaelzhiluo) [reply sky/backends/onprem_utils.py:72]: Done, thanks!

File: sky/backends/onprem_utils.py (hunk 4 above)
Comment (michaelzhiluo) [reply sky/backends/onprem_utils.py:82]: Done, thanks! Moved it to right after `yaml_config` is retrieved.

File: sky/backends/onprem_utils.py (hunk 5 above)
Comment (michaelzhiluo) [reply sky/backends/onprem_utils.py]: Done, thanks!

**Review by romilbhardwaj — APPROVED** [review]
lgtm, thanks for adding this! 🥳

**Pull request merged**